src/Vision/V1/ImageAnnotatorClient.php

```diff
@@ -30,13 +30,458 @@
+use InvalidArgumentException;
 
 /**
  * {@inheritdoc}
  */
 class ImageAnnotatorClient extends ImageAnnotatorGapicClient
 {
-    // This class is intentionally empty, and is intended to hold manual
-    // additions to the generated {@see ImageAnnotatorClientImpl} class.
+    use VisionHelpersTrait;
+
+    /**
+     * Creates an Image object that can be used as part of an image annotation request.
+     *
+     * Example:
+     * ```
+     * //[snippet=resource]
+     * $imageResource = fopen('path/to/image.jpg', 'r');
+     * $image = $imageAnnotatorClient->createImageObject($imageResource);
+     * $response = $imageAnnotatorClient->faceDetection($image);
+     * ```
+     *
+     * ```
+     * //[snippet=data]
+     * $imageData = file_get_contents('path/to/image.jpg');
+     * $image = $imageAnnotatorClient->createImageObject($imageData);
+     * $response = $imageAnnotatorClient->faceDetection($image);
+     * ```
+     *
+     * ```
+     * //[snippet=url]
+     * $imageUri = "gs://my-bucket/image.jpg";
+     * $image = $imageAnnotatorClient->createImageObject($imageUri);
+     * $response = $imageAnnotatorClient->faceDetection($image);
+     * ```
+     *
+     * @param  resource|string $imageInput An image to configure with
+     *         the given settings. This parameter will accept a resource, a
+     *         string of bytes, or the URI of an image in a publicly-accessible
+     *         web location.
+     * @return Image
+     * @throws InvalidArgumentException
+     */
+    public function createImageObject($imageInput)
+    {
+        return $this->createImageHelper(Image::class, ImageSource::class, $imageInput);
+    }
+
+    /**
+     * Run image detection and annotation for an image.
+     *
+     * Example:
+     * ```
+     * use Google\Cloud\Vision\V1\Feature;
+     * use Google\Cloud\Vision\V1\Feature_Type;
+     *
+     * $imageResource = fopen('path/to/image.jpg', 'r');
+     * $image = $imageAnnotatorClient->createImageObject($imageResource);
+     * $feature = new Feature();
+     * $feature->setType(Feature_Type::FACE_DETECTION);
+     * $features = [$feature];
+     * $response = $imageAnnotatorClient->annotateImage($image, $features);
+     * ```
+     *
+     * @param Image     $image        The image to be processed.
+     * @param Feature[] $features     Requested features.
+     * @param array     $optionalArgs {
+     *                                Optional.
+     *
+     *     @type ImageContext        $imageContext  Additional context that may accompany the image.
+     *     @type RetrySettings|array $retrySettings
+     *          Retry settings to use for this call. Can be a
+     *          {@see Google\ApiCore\RetrySettings} object, or an associative array
+     *          of retry settings parameters. See the documentation on
+     *          {@see Google\ApiCore\RetrySettings} for example usage.
+     * }
+     *
+     * @return AnnotateImageResponse
```

Comment:
I tested this out, and think it could be useful to encode the response back to a PHP array. It seems rather odd to hide most of the proto message complexity from users on the request side, but to force them to use the ApiCore Serializer or a bunch of getters to read the responses they receive.

Reply:
We made this exact mistake in Node -- learn from our mistake. :-) The problem is that there is (somewhat rarely used) additional information that is interesting in the response, and more may be added in the future.

Reply:
That makes sense. What about a helper method to serialize to array then? Could take any instance of `\Google\Protobuf\Internal\Message`, run it through `\Google\ApiCore\Serializer::encodeMessage()` and return an array? Just some way to save users from needing to understand the complexities of GAX or protos.

Reply:
I like the concept of reducing the complexity of the result, to take that thought one step further what if the protobuf message class had a toArray method?

Reply:
I'd be very happy to see that, but it would take modifications to Protobuf which I figured would be a much bigger lift than handling it in the gapics.

Reply:
They definitely would be! However, it feels like something we should push for and something like this could be a good argument as to why we need it sooner rather than later.

Reply:
I think that a toArray method would be a good thing to have in protobuf.

Re unpacking the response ourselves, I would like to avoid it if possible. My theory of when to provide helpers and when not to is: are we providing something helpful *other that* hiding protobuf? Because if all we are doing is hiding protobuf, then we should consider pushing on protobuf to solve the issue. In the case of the helpers to construct the Image object, I feel that the usefulness of accepted multiple types and navigating the structure appropriately makes this a useful addition, and is something that protobuf could never do on its own.